qt-core/src/qtcli/runner.ts

```diff
@@ -0,0 +1,116 @@
+// Copyright (C) 2024 The Qt Company Ltd.
+// SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only
+
+import * as os from 'os';
+import * as fs from 'fs/promises';
+import * as vscode from 'vscode';
+
+import { QtcliAction, openUri, logger, errorString } from '@/qtcli/common';
+
+export class QtcliRunner {
+  private _workingDir = os.homedir();
+  private _qtcliExecPath = '';
+  private _terminal: vscode.Terminal | undefined = undefined;
+  private _fsDisposables: vscode.Disposable[] = [];
+  private _terminalDisposables: vscode.Disposable[] = [];
+
+  dispose() {
+    if (this._terminal) {
+      this._terminal.dispose();
+      this._terminal = undefined;
+
+      for (const d of this._terminalDisposables) {
+        d.dispose();
+      }
+
+      this._terminalDisposables = [];
+      this._disposeFsWatcher();
+    }
+  }
+
+  private _disposeFsWatcher() {
+    for (const d of this._fsDisposables) {
+      d.dispose();
+    }
+
+    this._fsDisposables = [];
+  }
+
+  public setWorkingDir(dir: string) {
+    if (this._workingDir !== dir) {
+      this._workingDir = dir;
+      this.dispose();
+    }
+  }
+
+  public setQtcliExePath(fullPath: string) {
+    this._qtcliExecPath = fullPath;
+  }
+
+  public async run(action: QtcliAction, arg: string) {
+    try {
+      await fs.mkdir(this._workingDir, { recursive: true });
+      this._disposeFsWatcher();
+      this._setupFsWatcher(action, arg);
+
+      if (action === QtcliAction.NewProject) {
+        this._runQtcli(['new', arg]);
+      } else {
+        this._runQtcli(['new-file', arg]);
```

Comment:
NIT: Just note: This command doesn't work right now but it will be added here https://github.com/qt-labs/vscodeext/pull/5/files#diff-913a1777e689a2da7ed5bfd67cbfbf70bfed23980ef4e4eefd6b83260180336eR21